tests/test_target.py

```diff
@@ -146,5 +148,256 @@ def test_image_cache(self):
+        deep_pth = self._create_temp_file(prefix="item_for_deep", extension=".pth")
+        cascade_xml = self._create_temp_file(prefix="item_for_cascade", extension=".xml")
+        # no need to mock png files -- the Image target uses PIL.Image.save(), which we mocked
+
+        # destination stepfile
+        target_filename = self._create_temp_file(extension=".steps")
+
+        stepsfile_contents = [
+            "item_for_contour.png	some_contour_matchfile.match",
+            "item_for_tempfeat.png	some_tempfeat_matchfile.match",
+            "item_for_feature.png	some_feature_matchfile.match",
+            "{}	some_deep_matchfile.match".format(deep_pth),
+            "{}	some_cascade_matchfile.match".format(cascade_xml),
+            "item_for_template.png	some_template_matchfile.match",
+            "item_for_autopy.png	some_autopy_matchfile.match",
+            "{}	some_text_matchfile.match".format(os.path.splitext(text_file)[0]),
+            "some_text_content	some_text_matchfile.match"
+        ]
+
+        expected_content = [
+            "item_for_contour.png	item_for_contour.match",
+            "item_for_tempfeat.png	item_for_tempfeat.match",
+            "item_for_feature.png	item_for_feature.match",
+            "{0}.pth	{0}.match".format(os.path.splitext(deep_pth)[0]),
+            "{0}.xml	{0}.match".format(os.path.splitext(cascade_xml)[0]),
+            "item_for_template.png	item_for_template.match",
+            "item_for_autopy.png	item_for_autopy.match",
+            "{0}.txt	{0}.match".format(os.path.splitext(text_file)[0]),
+            "some_text_content	some_text_content.match"
+        ]
+
+        source_stepsfile = self._create_temp_file(prefix=self.stepsfile_name,
+            extension=".steps", contents=os.linesep.join(stepsfile_contents))
+
+        Path().add_path(os.path.dirname(text_file))
+        try:
+            chain = Chain(os.path.splitext(source_stepsfile)[0])
+            chain.save(target_filename)
+
+            with open(target_filename, "r") as f:
+                generated_content = f.read().splitlines()
+
+            # assert that the generated steps file has the expected content
+            self.assertEqual(generated_content, expected_content)
+
+            # build a list of the match filenames generated from
+            # the calls to `Finder.to_match_file()`
+            generated_match_names = []
+            for c in self.mock_match_write.call_args_list:
+               generated_match_names.append(c[0][1])
+
+            # get a list
+            expected_match_names = [x.split("\t")[1] for x in expected_content]
+            expected_match_names.insert(0, os.path.splitext(source_stepsfile)[0] + ".match")
+
+            # and assert that a match file was generated for each line
+            # and for the steps file itself
+            self.assertEqual(generated_match_names, expected_match_names)
+        finally:
+            Path().remove_path(os.path.dirname(text_file))
```

Comment:
Since these are unit tests it might be better to remove the directory directly and not relay on the tested functionality itself to do the cleanup. Unless you mean that this is tested too in which case it might still need extra backup cleanup.

Reply:
The `remove_path()` method just removes the path from the list of paths that are searched, it doesn't unlink it from the disk

Reply:
Fair enough